### PR TITLE
[VM]Add BGPSentinel testcase and testplan

### PR DESCRIPTION
**Pull request by guangyao6**
### Description of PR

Summary: Add BGPSentinel testcase and testplan
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
In SONiC, a new peer-group (BGPSentinel) is added in BGP configuration. This testcase is to verify that IBGP session would be established between DUT and BGPSentinel host (simulated by exabgp in ptf). And to verify exabgp in BGPSentinel host would advertise and withdraw route to DUT.
#### How did you do it?
1. Using t1-lag topology, and configure route in ptf to ensure that ptf and dut would ping each other.
2. Add BGPSentinel configuration in DUT and start exabgp in ptf to setup IBGP connection.
3. Advertise route from ptf with higher local-preference and no-export community to DUT, check route in DUT would be suppressed.
4. Withdraw route from ptf, check route in DUT would be advertised to EBGP peers.
#### How did you verify/test it?
Setup t1-lag topology. Using command "./run_tests.sh -m individual -a False -n vms-kvm-t1-lag -u  -c bgp/test_bgp_sentinel.py -f vtestbed.yaml -i veos_vtb -e "--neighbor_type=eos"" to run testcase. Testcase would pass.
#### Supported testbed topology if it's a new test case?
t1-lag

### CODE REVIEW

File: tests/bgp/test_bgp_sentinel.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+    duthost.shell("sonic-cfggen -j {} -w".format(BGPSENTINEL_CONFIG_FILE))
+
+    # Once https://github.com/sonic-net/sonic-buildimage/pull/14844 is merged into sonic 202205, we can remove this
+    duthost.shell("vtysh -c \"configure terminal\" -c \"ipv6 nht resolve-via-default\"")
+
+    yield lo_ipv4_addr, lo_ipv6_addr, spine_bp_addr, ptf_bp_v4, ptf_bp_v6
+
+    # Cleanup bgp monitor
+    duthost.run_sonic_db_cli_cmd("CONFIG_DB del 'BGP_SENTINELS|BGPSentinel'", asic_index='all')
+    duthost.run_sonic_db_cli_cmd("CONFIG_DB del 'BGP_SENTINELS|BGPSentinelV6'", asic_index='all')
```
Comment (jcaiMR) [inline tests/bgp/test_bgp_sentinel.py:172]: Will "CONFIG_DB del 'BGP_SENTINELS|BGPSentinel'" cause bgp container restart ? or we need specify it in script ?

File: tests/bgp/test_bgp_sentinel.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+    duthost.file(path=BGPSENTINEL_CONFIG_FILE, state='absent')
+
+
+@pytest.fixture(scope="module")
+def ptf_setup_teardown(dut_setup_teardown, rand_selected_front_end_dut, ptfhost, tbinfo):
+    duthost = rand_selected_front_end_dut
```
Comment (jcaiMR) [inline tests/bgp/test_bgp_sentinel.py:179]: Does this script need run in dual-tor scenario ? If yes, rand_selected_front_end_dut may select the lower tor, may need toggle all operation here.

**Review by yxieca — CHANGES_REQUESTED** [review]
Please schedule a time to present test plan in the test subgroup meeting.

File: tests/bgp/test_bgp_sentinel.py (hunk 2 above)
Comment (guangyao6) [reply tests/bgp/test_bgp_sentinel.py:179]: Thanks for the reminder. The script will only run in T1 scenario.

File: tests/bgp/test_bgp_sentinel.py (hunk 1 above)
Comment (guangyao6) [reply tests/bgp/test_bgp_sentinel.py:172]: It will not cause bgp container restart.

File: tests/bgp/test_bgp_sentinel.py (hunk 2 above)
Comment (guangyao6) [reply tests/bgp/test_bgp_sentinel.py]: Change fixture from rand_selected_front_end_dut to  rand_selected_dut

File: docs/testplan/BGP-BGPSentinel.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+The test is targeting a running SONIC system with fully functioning configuration. The purpose of the test is not to test specific API, but functional testing of BGPSentinel configuration on SONIC system.
+
+### Testbed
+The test will run on the following testbeds:
+* t1-lag (vs)
+
+## Setup configuration
+This test doesn't require any configuration in ansible deployment. IBGP session between BGPSentinel (BGPS) host and DUT are created in set up phase of this case and are cleaned in tear down phase of this case. BGPSentinel (BGPS) host is simulated by exabgp in PTF.
+
+## Test
```
Comment (bpar9) [inline docs/testplan/BGP-BGPSentinel.md:20]: - Please include the topology which you shared during community review in the testplan document

File: docs/testplan/BGP-BGPSentinel.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+The test will configure BGPSentinel session and then check that IBGP session would be setup between BGPSentinel (BGPS) host and DUT. After session set up, BGPSentinel (BGPS) will advertise routes with higher local-preference and no-export community. These routes will act as best-path in DUT, which would suppress routes received from t0. The test will cover both V4 and V6 BGPSentinel session.
+
+## Test cases
```
Comment (bpar9) [inline docs/testplan/BGP-BGPSentinel.md:23]: Based on what we discussed during community Testplan review, can you please include the following:
- have a test with bgp soft & hard reset triggers and verify that the DUT is still honoring the right best path from the BGPSentinel peer and suppressing it accordingly to T2 peers

**Review by bpar9 — CHANGES_REQUESTED** [review]

File: docs/testplan/BGP-BGPSentinel.md (hunk 3 above)
Comment (guangyao6) [reply docs/testplan/BGP-BGPSentinel.md:44]: The topology added to the testplan doc. Thanks.

File: docs/testplan/BGP-BGPSentinel.md (hunk 4 above)
Comment (guangyao6) [reply docs/testplan/BGP-BGPSentinel.md:47]: Add Test case  3 - Soft/Hard reset IBGP V4 and V6 Session in DUT to cover this part. Thanks.

**Review by wangxin — DISMISSED** [review]

**Review by bpar9 — APPROVED** [review]

**Review by yxieca — APPROVED** [review]

**Pull request merged**